lib/BoundaryValueDiffEqFIRK/src/firk.jl

```diff
@@ -444,43 +444,66 @@ function __construct_nlproblem(
     L = length(resid_bc)
     resid_collocation = __similar(y, cache.M * (N - 1) * (stage + 1))
 
-    loss_bcₚ = (iip ? __Fix3 : Base.Fix2)(loss_bc, cache.p)
-    loss_collocationₚ = (iip ? __Fix3 : Base.Fix2)(loss_collocation, cache.p)
+    bc_diffmode = if jac_alg.bc_diffmode isa AutoSparse
+        AutoSparse(get_dense_ad(jac_alg.bc_diffmode);
+            sparsity_detector = SparseConnectivityTracer.TracerLocalSparsityDetector(),
```

Comment:
are we sure that local sparsity detection is what we want here?

Reply:
The sparsity pattern of the boundary conditions is more of a local sparsity pattern since usually the boundary conditions are only evaluated in some fixed points

Reply:
Be careful then, the results of sparse preparation with local sparsity detection can only be reused if you can guarantee that the sparsity pattern doesn't change between calls